navigation menus: improve layout and styling for side- and topnav

jennur opened the pull request:
Closes https://github.com/inveniosoftware/invenio-administration/issues/14

- Adds styling to sidebar menu
- Enables sidebar to be toggled
    - Note: Currently toggles all the way, when we have icons for the menu items, it can be reiterated to toggle to show only icons

Needs styling from `invenio-theme`: 

## Screenshots

### Desktop
<img width="1508" alt="Screenshot 2022-08-11 at 15 45 20" src="https://user-images.githubusercontent.com/21052053/184155095-584ca968-2bee-4d9e-9de4-97e6b807b246.png">
<img width="1505" alt="Screenshot 2022-08-11 at 15 45 31" src="https://user-images.githubusercontent.com/21052053/184155084-135a0cea-7797-4c0d-8527-92e3b01bfa69.png">

### Tablet
<img width="984" alt="Screenshot 2022-08-11 at 15 45 55" src="https://user-images.githubusercontent.com/21052053/184155079-a8255e19-9e01-4422-a0c5-a19c127e9d6d.png">
<img width="983" alt="Screenshot 2022-08-11 at 15 46 04" src="https://user-images.githubusercontent.com/21052053/184155069-aa817316-57af-4955-b39d-8311ad55927e.png">
<img width="982" alt="Screenshot 2022-08-11 at 15 46 16" src="https://user-images.githubusercontent.com/21052053/184155064-9c1af535-79cf-4df4-9fd7-4feafcee0dcb.png">

### Mobile
<img width="403" alt="Screenshot 2022-08-11 at 16 33 07" src="https://user-images.githubusercontent.com/21052053/184158615-4a6bd60d-099c-4512-a35f-9930dd6fd9fa.png">
<img width="401" alt="Screenshot 2022-08-11 at 16 33 14" src="https://user-images.githubusercontent.com/21052053/184158607-6fa2d4b1-4b3b-4842-afd1-5a4d1e99e063.png">

NRodriguezcuellar commented on invenio_administration/assets/semantic-ui/js/invenio_administration/theme.js:
```diff
@@ -55,3 +55,32 @@ $invenioMenu.on("keydown", (event) => {
+const $rightIcon = $("button.side-menu-toggle .right.icon");
+const $sidebarMenu = $("#admin-side-menu");
+const $mainContent = $("#admin-main-content");
+const mainContentClassList = $mainContent[0].classList.value;
```
i am wondering, does it always return an array, even if you give it an ID?

jennur replied:
Hmm, it did in this case at least, which is why I needed to access it like that. But I can double check...

NRodriguezcuellar replied:
not exactly but something like [that](https://www.quora.com/Is-there-a-way-for-jQuery-to-return-a-single-object-instead-of-an-array-of-objects-For-example-if-I-use-an-id-selector) apparently.

If we would want to reduce jquery i think it could be done without too much added complexity using the [classList api](https://www.w3schools.com/jsref/prop_element_classlist.asp) but not sure if that is a consideration at the moment.

jennur replied:
Will check it out!

Just for some context: The reason I used `classList` is that it returns the full list, with the multiple occurrences of some classes. With semantic ui, we need to add e.g. the 'wide' class several times to get the responsiveness for each screen, for instance `sixteen wide mobile twelve wide tablet eight wide computer`. The `removeClass()` and `addClass()` functions treats multiple occurrences as only _one_ class, i.e. using `$elem.addClass("sixteen wide mobile twelve wide tablet")` will only add the `wide` class once.

kpsherva replied:
I think it is not necessary for now

NRodriguezcuellar reviewed (approved):
Looks good!

jennur commented:
## Updates
- Add page title to top bar
- *Note: I did not capitalize the strings, as it was discussed this should be done by the user from the config*

![Screenshot 2022-08-24 at 10 30 48](https://user-images.githubusercontent.com/21052053/186370307-368977cc-0111-48bd-959a-c09dff6324b4.png)
![Screenshot 2022-08-24 at 10 35 24](https://user-images.githubusercontent.com/21052053/186371393-c963807d-f0de-4149-b71f-3acd9cdc18a8.png)

kpsherva commented:
> ## Updates
> * Add page title to top bar
> * _Note: I did not capitalize the strings, as it was discussed this should be done by the user from the config_
> 
> ![Screenshot 2022-08-24 at 10 30 48](https://user-images.githubusercontent.com/21052053/186370307-368977cc-0111-48bd-959a-c09dff6324b4.png) ![Screenshot 2022-08-24 at 10 35 24](https://user-images.githubusercontent.com/21052053/186371393-c963807d-f0de-4149-b71f-3acd9cdc18a8.png)

I really like the dividers in the menu, they are neat, and overall styles are getting cleaned. I have only two comments:

I think placing the title on the top navigation bar causes mixing the content, the top bar should focus on administration profile, and be separated from the actual content of the page (ie the title)
see examples: 
![image](https://user-images.githubusercontent.com/38131488/186414483-8db4a1e0-01a8-4f89-bed9-07270c41b048.png)
![image](https://user-images.githubusercontent.com/38131488/186414582-8c6c2dcf-3a22-44c7-b01d-c696528b2af0.png)
I would prefer a clear separation of the content and navigation areas as in the given examples

Could we move the menu collapse button lower to the actual menu area? Otherwise it is not completely clear to which panel the button refers to, and in addition I think it should be more "discrete" - maybe without the outline, and smaller than the RDM logo?

jennur commented:
## Updated screenshot
- Removed headline
- Changed toggle button to chevron icon
![Screenshot 2022-08-26 at 12 03 57](https://user-images.githubusercontent.com/21052053/186880867-442fe483-8568-41d5-8d23-5878a302f223.png)
